public/assets/panel/panel-statistics.js

```diff
@@ -1,124 +1,163 @@
+
+    yLabel =  $(".select[name='y-axis'] ul li:visible")[0].getAttribute('data-value');
+    yType =  $(".select[name='y-axis'] ul li:visible")[0].getAttribute('data-type');
+    yText =  $(".select[name='y-axis'] ul li:visible")[0].innerHTML;
+
+    districtLabel = $(".select[name='district'] ul li:visible")[0].getAttribute('data-value');
+
+    colorLabel = $(".select[name='color'] ul li:visible")[0].getAttribute('data-value');
+
+    $(".plotOption").removeClass("d-none");
+
+    if(xType=="none"){
+        $(".plotOption").addClass("d-none");
+        alert('Please select the x-axis');
+    }
+    if(yType=="none"){
+        $(".plotOption").addClass("d-none");
+        alert('Please select the y-axis');
+    }
+    if(xType=="quantitative"){
+        $(".plotOption > #barChart").parent().addClass("d-none");
+        $(".plotOption > #boxplot").parent().addClass("d-none");
+    }
+    if(xType=="qualitative"){
+        $(".plotOption > #scatter").parent().addClass("d-none");
+    }
+    if(yLabel=="BuildingNumber" || yLabel=="BuildingNumberPerKm2"){
+        $(".plotOption > #scatter").parent().addClass("d-none");
+        $(".plotOption > #boxplot").parent().addClass("d-none");
+    }
+    if(yLabel!="BuildingNumber" && yLabel!="BuildingNumberPerKm2"){
+        $(".plotOption > #barChart").parent().addClass("d-none");
+    }
 })
 
 $("#barChart").click(()=>{
 
-    if(chart){chart.destroy()}
-
-    data_bar = {
-        labels: ['Mong Kok', 'Central', 'Wan Chai', 'Sha Tin', 'Admiralty', 'Sai Ying Pun'],
-        datasets: [{
-            label: '# of Housing Estates',
-            data: [12, 19, 3, 5, 2, 3],
-            backgroundColor: [
-                'rgba(255, 99, 132, 0.2)',
-                'rgba(54, 162, 235, 0.2)',
-                'rgba(255, 206, 86, 0.2)',
-                'rgba(75, 192, 192, 0.2)',
-                'rgba(153, 102, 255, 0.2)',
-                'rgba(255, 159, 64, 0.2)'
-            ],
-            borderColor: [
-                'rgba(255, 99, 132, 1)',
-                'rgba(54, 162, 235, 1)',
-                'rgba(255, 206, 86, 1)',
-                'rgba(75, 192, 192, 1)',
-                'rgba(153, 102, 255, 1)',
-                'rgba(255, 159, 64, 1)'
-            ],
-            borderWidth: 1
-        }]
-    }
+    if(panelChart){panelChart.destroy()};
+    
+    $.getJSON('./assets/panel/data/buildingAroundMTR.json',(json)=>{
 
-    chart = new Chart("statistics-modal-chart", {
-        type: 'bar',
-        data: data_bar,
-        options: {
-            scales: {
-                x:{
-                    title:{
-                        display: true,
-                        text: xElement.innerHTML
-                    }
-                },
-                y: {
-                    title:{
-                        display: true,
-                        text: yElement.innerHTML
+        i = 1;
+        labels = Array(5).fill().map(()=>`${i}km around the MTR station`);
```

Comment:
```suggestion
        labels = Array(5).fill().map((_, i)=>`${i+1}km around the MTR station`);
```
It always displays 1km, so u may consider this change